### PR TITLE
Ms2 add remove book

### PR DESCRIPTION
**Pull request by Bishoy-Samwel**
- [x] Configure the Redux Store and importing the necessary methods from Redux
 *  Import createStore
*  Import combineReducers 
*  Import applyMiddleware 
*  Import your booksReducer,
- [x]  Write your book's actions and reducer using the ducks pattern. <https://github.com/erikras/ducks-modular-redux>
*  Define action types:
*  Set initial stateto be an empty array of books
*  Export your Action Creators (functions, which return actions):
*  Write your reducer and export it as default

- [x] In your React component responsible for adding new books set the data inputs in the local React state (set title and author ). Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action
- [x] In your React component responsible for removing books - implement that even by dispatching a corresponding action.

### CODE REVIEW

File: src/components/AddBookForm.js [hunk 1]
```diff
@@ -1,11 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
```
Comment (Emmyn5600) [inline src/components/AddBookForm.js:4]: - [x] I had an issue while trying to run the project locally. Please fix it.
uuid

File: src/components/Book.js [hunk 2]
```diff
@@ -1,20 +1,23 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { removeBook } from '../redux/books/books';
 
-export default function User(props) {
+export default function Book(props) {
```
Comment (Emmyn5600) [inline src/components/Book.js]: - [x] kindly use ES6 syntax. and make sure that you changed all you used the `function` to const as well as the arrow function.

File: src/redux/books/books.js [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
-  return { type: actionName };
-}
+export const addBook = payload => ({
+  type: ADD_BOOK,
+  payload,
+});
+export const removeBook = payload => ({
+  type: REMOVE_BOOK,
+  payload,
+});
```
Comment (Emmyn5600) [inline src/redux/books/books.js]: - [x] Let us create the actions before anything and call the actions created in the reducers, refer to the below example.

const ADD_BOOK = 'bookStore/books/ADD_BOOK';
const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';

const initialState = [];

export const addBook = (payload) => ({
  type: ADD_BOOK,
  payload,
});

export const removeBook = (payload) => ({
  type: REMOVE_BOOK,
  payload,
});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_BOOK:
      return [...state, action.payload];
    case REMOVE_BOOK:
      return state.filter((book) => book.id !== action.payload.id);

    default:
      return state;
  }
};

export default reducer;

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi, @Bishoy-Samwel 

Good job so far! 👏🏼 
Great work on making the project requirements :clap:
You've done well-implementing project requirements, but there are still some issues which aren't been addressed yet 
 
## STATUS: CHANGES REQUESTED :recycle:


### Changes requested.

- [x] Check the comment under the review.

 #### Optional suggestions
 
 - N/A
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by dasileker — APPROVED** [review]
## **Hi @Bishoy-Samwel** 

### **Optional**

- Please make sure to update the readme file, before sending the **_PR_** for review.

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 

Approved

 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**